cmd/kubebuilder/initproject/init.go

```diff
@@ -78,9 +77,24 @@ kubebuilder init repo --domain mydomain
 }
 
 func (o *initOptions) runInitRepo() {
-	version := runtime.Version()
-	if versionCmp(version, "go1.10") < 0 {
-		log.Fatalf("The go version is %v, must be 1.10+", version)
+	cmd := exec.Command("go", "version")
+	out, err := cmd.Output()
+	if err != nil {
+		log.Fatalf("Could not execute 'go version': %v", err)
+	}
+
+	goVersion := strings.TrimPrefix(strings.Split(string(out), " ")[2], "go")
+	if ver, err := semver.NewVersion(goVersion); err != nil {
+		if err != nil {
+			log.Fatalf("Invalid go version %q: %v", goVersion, err)
+		}
+		c, err := semver.NewConstraint(">= 1.10")
+		if err != nil {
+			log.Fatal("Invalid constraint: %v", err)
+		}
+		if !c.Check(ver) {
+			log.Fatalf("The go version is %v, must be 1.10+", goVersion)
+		}
```

Comment:
Can we abstract the above code into a separate function ?

Reply:
ack